src/main/php/de/thekid/dialog/Signing.php

```diff
@@ -0,0 +1,34 @@
+<?php namespace de\thekid\dialog;
+
+use text\hash\{Algorithm, Hashing};
+use util\Secret;
+
+/** @test de.thekid.dialog.unittest.SigningTest */
+class Signing {
```

Comment:
This class takes care of preventing people from misusing `/statistics/{slug}`. The links generated in the frontend's JavaScript code are valid for 30 seconds; the API discards invalid requests silently

Reply:
Quoting https://stackoverflow.com/a/30638622 (*which is related to AWS S3, but is valid here too*):

> A Signed URL is safe because:
> 
> * It is valid for only a limited time period that you specify
> * It is valid only for the [...] object that you specify
> * It cannot be used to retrieve a different object nor can the time period be modified (because it would invalidate the signature)
> 
> However, anyone can use the URL during the valid time period. So, if somebody Tweets the URL, many people could potentially access the object until the expiry time.

Reply:
We could also use one-time links with [MongoDB time series](https://www.mongodb.com/docs/manual/core/timeseries-collections/), but I'll leave this for another PR.